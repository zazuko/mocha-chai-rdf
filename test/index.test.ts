/* eslint-disable @typescript-eslint/no-explicit-any */
import * as url from 'node:url'
import util from 'node:util'
import * as chai from 'chai'
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot'
import { glob } from 'glob'
import { expect } from 'chai'
import Mocha from 'mocha'

const __dirname = url.fileURLToPath(new URL('.', import.meta.url))

function sanitizeOutput(output: string): string {
  return output
    .replace(/\x1b\[[0-9;]*m/g, '') // eslint-disable-line no-control-regex
    .replace(/(\d+ms)/g, '0ms')
    .replace(/at (.*) \(.+?\)/g, 'at $1')
}

describe('mocha-chai-rdf', () => {
  chai.use(jestSnapshotPlugin())

  const files = glob.sync('tests/*.ts', {
    cwd: __dirname,
  })

  it('test suites', async () => {
    let output = ''

    ;(Mocha.reporters.Base as any).consoleLog = (...args: unknown[]) => {
      output += util.format(...args) + '\n'
    }

    const mocha = new Mocha()
    files.forEach((file) => {
      mocha.addFile(`./test/${file}`)
    })
    await mocha.loadFilesAsync()
    mocha.reporter(Mocha.reporters.List)

    await new Promise<void>((resolve) => {
      mocha.run(() => {
        // eslint-disable-next-line no-console
        ;(Mocha.reporters.Base as any).consoleLog = console.log
        resolve()
      })
    })

    expect(sanitizeOutput(output)).toMatchSnapshot()
  })
})
