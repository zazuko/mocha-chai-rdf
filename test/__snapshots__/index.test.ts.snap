// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocha-chai-rdf test suites 1`] = `
"<http://example.org/> <http://example.org/> <http://example.org/> .
"
`;

exports[`mocha-chai-rdf test suites 2`] = `"[{\\"@id\\":\\"http://example.org/s\\",\\"http://example.org/p\\":{\\"@id\\":\\"http://example.org/o\\"}}]"`;

exports[`mocha-chai-rdf test suites 3`] = `
"
  ✔ store.js createStore before, turtle loads expected triples: 0ms
  ✔ store.js createStore before, turtle provides access properties: 0ms
  ✔ store.js createStore beforeEach, turtle loads expected triples: 0ms
  ✔ store.js createStore beforeEach, trig graph exists loads expected triples: 0ms
  1) store.js createStore beforeEach, trig graph does not exist throws
  ✔ store.js createStore beforeEach, trig, all graphs loads expected triples: 0ms
  ✔ store.js createEmpty initializes empty dataset: 0ms
  ✔ store.js createEmpty sparql updates are reflected in dataset: 0ms
  ✔ store.js createEmpty provides access properties: 0ms
  ✔ sparql-clients.js parsingClient can be queried with SELECT: 0ms
  ✔ sparql-clients.js parsingClient can be queried with CONSTRUCT: 0ms
  ✔ sparql-clients.js parsingClient can be queried with ASK: 0ms
  ✔ sparql-clients.js parsingClient can be updated: 0ms
  ✔ sparql-clients.js sparqlClient can be queried with SELECT: 0ms
  ✔ sparql-clients.js sparqlClient can be queried with CONSTRUCT: 0ms
  ✔ sparql-clients.js sparqlClient can be queried with ASK: 0ms
  ✔ sparql-clients.js sparqlClient can be updated: 0ms
  ✔ snapshots.js can be used to match canonical representation: 0ms
  ✔ snapshots.js preserves original functionality: 0ms
  ✔ matchers.js term can compare oxigraph with @rdfjs: 0ms
  ✔ matchers.js term can compare oxigraph with rdf-data-factory: 0ms
  ✔ matchers.js term can compare @rdfjs with rdf-data-factory: 0ms
  ✔ matchers.js term when actual object is pointer succeeds when equal: 0ms
  2) matchers.js term when actual object is pointer fails when not equal
  3) matchers.js term when actual object is pointer fails when multi-pointer
  ✔ matchers.js term when actual and expected objects are pointers succeeds when equal: 0ms
  ✔ matchers.js term when actual and expected objects are pointers succeeds when equal multi-pointers: 0ms
  4) matchers.js term when actual and expected objects are pointers fails when not equal multi-pointers

  24 passing (0ms)
  4 failing

  1) store.js
       createStore
         beforeEach, trig
           graph does not exist
             throws:

      AssertionError [ERR_ASSERTION]: Test data not found in GRAPH <createStore/beforeEach,-trig/graph-does-not-exist>
      + expected - actual

      -false
      +true
      
      at assertNotEmpty
      at Context.get
      at Context.<anonymous>
      at process.processImmediate

  2) matchers.js
       term
         when actual object is pointer
           fails when not equal:

      AssertionError: expected '\\"http://example.org/\\"' but got 'a pointer to <http://example.org/>'
      + expected - actual

      -a pointer to <http://example.org/>
      +\\"http://example.org/\\"
      
      at Context.<anonymous>
      at process.processImmediate

  3) matchers.js
       term
         when actual object is pointer
           fails when multi-pointer:

      AssertionError: expected '<http://example.org/>' but got 'a pointer to <http://example.org/1>, …'
      + expected - actual

      -a pointer to <http://example.org/1>, <http://example.org/2>
      +<http://example.org/>
      
      at Context.<anonymous>
      at process.processImmediate

  4) matchers.js
       term
         when actual and expected objects are pointers
           fails when not equal multi-pointers:

      AssertionError: expected 'a pointer to <http://example.org/1>, …' but got 'a pointer to <http://example.org/1>, …'
      + expected - actual

      -a pointer to <http://example.org/1>, <http://example.org/2>
      +a pointer to <http://example.org/1>, <http://example.org/3>
      
      at Context.<anonymous>
      at process.processImmediate



"
`;
